tfhe/src/integer/server_key/radix_parallel/tests_unsigned/mod.rs

```diff
@@ -142,7 +144,30 @@ fn panic_if_any_block_info_exceeds_max_degree_or_noise(
     max_degree: MaxDegree,
     max_noise_level: MaxNoiseLevel,
 ) {
-    for (i, block) in ct.blocks.iter().enumerate() {
+    if ct.blocks.is_empty() {
+        return;
+    }
+
+    // The max degree is made such that a block is able to receive the carry from
+    // its predecessor when using the sequential propagation algorithm.
+    //
+    // However, as the first block does not have a predecessor, its max degree is actually
+    // bigger
+    let first_block = &ct.blocks[0];
+    let first_block_max_degree =
+        MaxDegree::new((first_block.message_modulus.0 * first_block.carry_modulus.0) - 1);
```

Comment:
I think we have a way to generate the max degree from the message_modulus and carry_modulus

It's `from_msg_carry_modulus`